src/v2/views/ClaimCollectionRewardsOverlay/ClaimContent.tsx

```diff
@@ -42,14 +62,14 @@ function ClaimContent({ data, onActionTxId, rewards, tip }: ClaimContentProps) {
   );
 
   useEffect(() => {
-    if (hasMultipleAvatars) return;
+    if (hasMultipleAvatars || !isOpen) return;
```

Comment:
What's the intent of this?

Reply:
It is a bit complicated. Before the change, the ClaimContent was rendered only when a user clicked the 'Get Rewards' button, and the overlay was open. However, because this behaviour has changed the ClaimContent now renders regardless of 'Get Rewards' has been clicked or not, potentially creating invalid transactions by this `useEffect` statement.

Reply:
> However, because this behaviour has changed the ClaimContent now renders regardless of 'Get Rewards' has been clicked or not, potentially creating invalid transactions by this useEffect statement.

I have no context about these changes, is it just for V2?

Reply:
If then, can you explain more about those changes?

Reply:
Yes, it is specific to V2 only. The v1 behaviour wouldn't be affected.

Reply:
I mean, do users not need the claim button (aka `Get Rewards` button) to click anymore in V2? I don't have a context about that.

Reply:
When the `OverlayBase` (which is a fancy mechanism for displaying a dialog on V2) is rendered in [`ClaimCollectionRewardsOverlay`](https://github.com/planetarium/9c-launcher/pull/1246/files#diff-9508a74246960570acb5cc90b21479291632541e102d8852defa7e4849a93e62) directly, the component doesn't know whether the user has multiple avatars and showed dialog anyway, which resulted in empty dialog flickering if the user has only one avatar.

However, the `ClaimContent` knows whether the user has multiple avatars, so if it controls whether the dialog should be shown or not, it can hide the dialog when it is unnecessary.

Reply:
Hmm.. then in this comment

> However, because this behaviour has changed the ClaimContent now renders regardless of 'Get Rewards' has been clicked or not, potentially creating invalid transactions by this useEffect statement.

transactions are just a typo about the transition? I feel like that in my understanding.

Reply:
It is kinda both, I forgot to mention it. Because the dialog is now rendered inside ClaimContent the component is rendered regardless of it's opened or not, causing to send tx if there's only one avatar.

Reply:
Understand